test_integration/mosaic-create/mosaic-create.test.ts

```diff
@@ -62,6 +62,12 @@ describe('Mosaic create', () => {
     Shell.executeInShell(command, { stdio: 'inherit' });
   });
 
+
+  it('Deploy stake pool contract with mosaic config option', async () => {
+    const command = `./mosaic setup-stake-pool ${originChainId} ${originWeb3RPCEndPoint} ${originDeployerAddress} ${originDeployerAddress} ${originDeployerAddress} --mosaic-config ~/.mosaic/dev-origin/mosaic.json`;
```

Comment:
these tests could be made stronger to catch potential bug which we mentioned above. May be by copying over the contents to a new temp file. Then deploying and verifying states of both the files.